nav2_mppi_controller/src/controller.cpp

```diff
@@ -122,6 +130,36 @@ void MPPIController::setSpeedLimit(const double & speed_limit, const bool & perc
   optimizer_.setSpeedLimit(speed_limit, percentage);
 }
 
+void MPPIController::publish_optimal_path(const xt::xtensor<float, 2> & optimal_traj)
+{
+  auto & size = optimal_traj.shape()[0];
+  if (!size) {
+    return;
+  }
+
+  nav_msgs::msg::Path optimal_path;
+  optimal_path.header.stamp = clock_->now();
```

Comment:
Use the stamp from the TwistStamped returned so that they are properly synchronized